platforms/core-runtime/internal-instrumentation-processor/src/test/groovy/org/gradle/internal/instrumentation/extensions/property/PropertyUpgradeCustomInterceptorCodeGenTest.groovy

```diff
@@ -125,32 +131,67 @@ class PropertyUpgradeCustomInterceptorCodeGenTest extends InstrumentationCodeGen
+                            return true;
+                        }
+                        if (name.equals("maxErrors") && descriptor.equals("()I") && (opcode == Opcodes.INVOKEVIRTUAL || opcode == Opcodes.INVOKEINTERFACE)) {
+                            mv._INVOKESTATIC(\$\$_BRIDGE_FOR\$\$_TASK\$\$_MAX_ERRORS_ADAPTER_TYPE, "access_get_maxErrors", "(Lorg/gradle/test/Task;)I");
+                            return true;
+                        }
+                        if (name.equals("maxErrors") && descriptor.equals("(I)Lorg/gradle/test/Task;") && (opcode == Opcodes.INVOKEVIRTUAL || opcode == Opcodes.INVOKEINTERFACE)) {
+                            mv._INVOKESTATIC(\$\$_BRIDGE_FOR\$\$_TASK\$\$_MAX_ERRORS_ADAPTER_TYPE, "access_set_maxErrors", "(Lorg/gradle/test/Task;I)Lorg/gradle/test/Task;");
+                            return true;
+                        }
+                        if (name.equals("setMaxErrors") && descriptor.equals("(I)V") && (opcode == Opcodes.INVOKEVIRTUAL || opcode == Opcodes.INVOKEINTERFACE)) {
+                            mv._INVOKESTATIC(\$\$_BRIDGE_FOR\$\$_TASK\$\$_MAX_ERRORS_ADAPTER_TYPE, "access_set_setMaxErrors", "(Lorg/gradle/test/Task;I)V");
+                            return true;
+                        }
+                    }
+                    return false;
+            }
+        """
+        // Just make sure that the groovy interceptors are generated
+        def expectedGroovyInterceptors = source """
+            package org.gradle.internal.classpath.generated;
+            @Generated
+            public class InterceptorDeclaration_PropertyUpgradesGroovyInterceptors_TestProject {
+                @Generated
+                public static class GetMaxErrorsCallInterceptor extends AbstractCallInterceptor implements SignatureAwareCallInterceptor, FilterableCallInterceptor, FilterableBytecodeInterceptor.BytecodeUpgradeInterceptor, PropertyAwareCallInterceptor {
+                    public GetMaxErrorsCallInterceptor() {
+                        super(InterceptScope.readsOfPropertiesNamed("maxErrors"), InterceptScope.methodsNamed("getMaxErrors"));
+                    }
+                }
+                @Generated
+                public static class MaxErrorsCallInterceptor extends AbstractCallInterceptor implements SignatureAwareCallInterceptor, FilterableCallInterceptor, FilterableBytecodeInterceptor.BytecodeUpgradeInterceptor {
+                    public MaxErrorsCallInterceptor() {
+                        super(InterceptScope.methodsNamed("maxErrors"));
+                    }
+                }
+                @Generated
+                public static class SetMaxErrorsCallInterceptor extends AbstractCallInterceptor implements SignatureAwareCallInterceptor, FilterableCallInterceptor, FilterableBytecodeInterceptor.BytecodeUpgradeInterceptor {
+                    public SetMaxErrorsCallInterceptor() {
+                        super(InterceptScope.methodsNamed("setMaxErrors"));
```

Comment:
❓ again, missing `InterceptScope.writesOfPropertiesNamed`

Reply:
I'll try to add that and see what happens, thanks!